x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/hooks/use_has_index_permission.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+
+import { useMlKibana } from '../../../../contexts/kibana';
+
+export const useHasRequiredIndicesPermissions = (indexName: string) => {
+  const [hasIndexPermissions, setHasIndexPermissions] = useState<boolean>(false);
+  const {
+    services: {
+      mlServices: {
+        mlApiServices: { hasPrivileges },
+      },
+    },
+  } = useMlKibana();
+
+  useEffect(
+    function checkRequiredIndexPermissions() {
+      async function checkPrivileges() {
+        const privileges = await hasPrivileges({
+          index: [
+            {
+              names: [indexName],
+              privileges: ['create_index', 'manage', 'index', 'read'],
+            },
+          ],
+        });
+
+        setHasIndexPermissions(
+          privileges.hasPrivileges === undefined ||
```

Comment:
is this logic correct?
if `privileges.hasPrivileges` is not defined, `hasIndexPermissions` will be true.

Reply:
Good catch! Fixed in 51e4a289dfc2e1810a7a3ab1d28febae50c405cc